### ISSUE TITLE
Broken on latest canary

### ISSUE THREAD

**Issue by trentmwillis** [dgeb/ember-engines#76]
Looks like something broke for this addon when using the latest version of Ember Canary. Components aren't rendered within an Engine, but no error is given. I would've opened an issue on Ember proper, but I have no clue what the root cause is.

**Comment by rwjblue** [comment]
Awesome, thank you for reporting. I'll to dig into tonight/tomorrow morning.

**Comment by trentmwillis** [comment]
Out of curiosity, how important is it to run on Canary? Since this addon imports the `Engine` and other classes using `emberRequire` everything works fine on the current Release or Beta builds. Is the Canary recommendation simply to acknowledge that this has some cutting edge APIs?

**Comment by dgeb** [comment]
@trentmwillis We are recommending Canary until this addon only uses Ember's public APIs. Until then, we'll be moving any custom overrides from this addon into Ember Core gradually, so we can only recommend use of Ember Canary for Ember Engines Canary. With that said, there's a good chance that particular versions of this addon will end up being compatible with particular Beta and Release builds of Ember itself.

**Comment by cmcdaniels** [comment]
@dgeb 

Any updates? Cloned the  [ember-engine-demo](https://github.com/dgeb/ember-engines-demo) repo and upon running ember s I get the following error:

Uncaught Error: Could not find module ember-views/views/select

**Comment by dgeb** [comment]
@cmcdaniels Yeah, there have been quite a few changes in Ember Canary lately. I started working on an upgrade to this addon a few days ago, but it's not quite green yet.

**Comment by trentmwillis** [comment]
I looked into this a bit more, in hopes of unblocking some efforts to use a newer version of canary and it seems there are two core issues:

1. [Outlets now need a template registered](https://github.com/emberjs/ember.js/blob/34c32a47d8aed7ba55d6ae529b5c921e38dd53c1/packages/ember-application/lib/system/application.js#L1161) otherwise they don't render anything, and
2.  [Some registrations are now "private"](https://github.com/emberjs/ember.js/blob/34c32a47d8aed7ba55d6ae529b5c921e38dd53c1/packages/ember-application/lib/system/application.js#L1090-L1093) which means looking them up by the normal string fails.

Unfortunately, looks like both of these require knowledge of truly private constructs in Ember, so I only verified fixes by monkeypatching.

**Comment by rwjblue** [comment]
Awesome, thank you @trentmwillis!

---

The first one can be solved with something like this around [here](https://github.com/dgeb/ember-engines/blob/master/addon/-private/engine-ext.js#L17):

```
const topLevelViewTemplate = emberRequire('ember-htmlbars/templates/top-level-view');
```

then registering around [here](https://github.com/dgeb/ember-engines/blob/master/addon/-private/engine-ext.js#L23-L85).

---

The second can be solved via annoyingly grabbing the private symbol and registering:

```
const P = emberRequire('container/register', 'privatize');
```

And registering the same...

**Comment by trentmwillis** [comment]
Ah cool, forgot you can use the `emberRequire` to get internal packages. I'll try to put together a PR.

**Comment by rwjblue** [comment]
Shhh, don't tell anyone :wink:

**Comment by rwjblue** [comment]
Fixed by #83 and #82
